### PR TITLE
Refactor of ResNet model generation.

### PR DESCRIPTION
**Pull request by hgaiser**
This PR removes some duplicate code for generating resnet models.

Also, the name of the function was changed from `ResNet50RetinaNet` to `resnet50_retinanet`.

### CODE REVIEW

File: keras_retinanet/models/resnet.py [hunk 1]
```diff
@@ -19,66 +19,69 @@
...
+resnet_filename = 'ResNet-{}-model.keras.h5'
+resnet_resource = 'https://github.com/fizyr/keras-models/releases/download/v0.0.1/{}'.format(resnet_filename)
 
 custom_objects = retinanet.custom_objects.copy()
 custom_objects.update(keras_resnet.custom_objects)
 
 
-def ResNet50RetinaNet(inputs, num_classes, weights='imagenet', *args, **kwargs):
-    image = inputs
+def resnet_retinanet(num_classes, backbone='50', inputs=None, weights='imagenet', **kwargs):
```
Comment (de-vri-es) [inline keras_retinanet/models/resnet.py]: `backbone` as string seems a bit weird. Why not just use a the numbers 50, 101 and 152?

File: keras_retinanet/models/resnet.py [hunk 2]
```diff
@@ -19,66 +19,69 @@
...
+    if backbone not in allowed_backbones:
+        raise ValueError('Backbone (\'{}\') not in allowed backbones ({}).'.format(backbone, allowed_backbones))
 
-    # load pretrained imagenet weights?
+    # choose default input
+    if inputs is None:
+        inputs = keras.layers.Input(shape=(None, None, 3))
+
+    # determine which weights to load
     if weights == 'imagenet':
```
Comment (de-vri-es) [inline keras_retinanet/models/resnet.py]: I suppose this isn't new behaviour, but this seems like a weird place to download the weights. It also means that you can't use a file called 'imagenet' as weights. Not a huge problem, but it is a bit dirty.

Personally I'd put the downloading in a separate function that a script can invoke before creating the model. That would leave this function a lot cleaner.

File: keras_retinanet/models/resnet.py [hunk 3]
```diff
@@ -19,66 +19,69 @@
...
         weights_path = keras.applications.imagenet_utils.get_file(
-            'ResNet-50-model.keras.h5',
-            WEIGHTS_PATH_NO_TOP_50, cache_subdir='models', md5_hash='3e9f4e4f77bbe2c9bec13b53ee1c2319'
+            filename,
+            resource,
+            cache_subdir='models',
+            md5_hash=checksum
         )
+    elif weights == 'none':
+        weights_path = None
```
Comment (de-vri-es) [inline keras_retinanet/models/resnet.py]: I wouldn't allow `'none'` as string. Like `weights='imagenet'` it claims another valid filename as a special case. I'd force the user to pass a real `None` instead.

In my opinion, special handling for strings is something better done in argument parsing of a script than in API functions.

File: keras_retinanet/models/resnet.py (hunk 1 above)
Comment (hgaiser) [reply keras_retinanet/models/resnet.py]: It used to be `'resnet50'` but I changed that later on to `'50'`. But you're right, I'll change it.

File: keras_retinanet/models/resnet.py (hunk 3 above)
Comment (hgaiser) [reply keras_retinanet/models/resnet.py]: I like it :+1:

File: keras_retinanet/models/resnet.py (hunk 2 above)
Comment (hgaiser) [reply keras_retinanet/models/resnet.py]: Fair enough.

**Comment by hgaiser** [comment]
@de-vri-es I made the requested changes.

**Comment by de-vri-es** [comment]
Looks good, thanks.

**Comment by de-vri-es** [comment]
How's the backwards compatibility coming along? Are you still working on that?

**Comment by hgaiser** [comment]
Working is a big word, but yeah. Awaiting tests now, then it can get merged.

**Pull request merged**